library/src/main/java/org/apache/apex/malhar/lib/window/accumulation/AbstractPojoJoin.java

```diff
@@ -0,0 +1,258 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.lib.window.accumulation;
+
+import java.lang.reflect.Field;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.apex.malhar.lib.window.MergeAccumulation;
+import org.apache.commons.lang3.ClassUtils;
+
+import com.google.common.base.Throwables;
+
+import com.datatorrent.lib.util.KeyValPair;
+import com.datatorrent.lib.util.PojoUtils;
+
+/**
+ * Join Accumulation for Pojo Streams.
+ *
+ */
+public abstract class AbstractPojoJoin<InputT1, InputT2>
+    implements MergeAccumulation<InputT1, InputT2, List<List<Map<String, Object>>>, List<?>>
+{
+  protected final String[] keys;
+  protected final Class<?> outClass;
+  private transient List<KeyValPair<String,PojoUtils.Getter>>  gettersStream1;
+  private transient List<KeyValPair<String,PojoUtils.Getter>>  gettersStream2;
+  private transient List<KeyValPair<String,PojoUtils.Setter>>  setters;
+  public transient Set<String> keySetStream2;
+  public transient Set<String> keySetStream1;
+
+  public AbstractPojoJoin()
+  {
+    keys = new String[]{};
+    outClass = null;
+  }
+
+  public AbstractPojoJoin(Class<?> outClass, String... keys)
+  {
+    if (keys.length % 2 != 0) {
```

Comment:
Thank you Chinmay for this input, I will create a separate Jira under APEXMALHAR-2413 for this.